rts/motoko-rts/src/gc/mark_compact/bitmap.rs

```diff
@@ -2,55 +2,126 @@ use crate::mem_utils::memzero;
 use crate::memory::{alloc_blob, Memory};
 use crate::types::{size_of, Blob, Bytes, Obj};
 
+/* How the Wasm-heap maps to the bitmap
+
+  +---- RTS stack ----+---- Motoko statics ----+---- Dynamic heap --~~--+ Heap limit
+                  (prefix words)                   bitmap lives here -> | BM |
+                                                                       /      \
+                                                                      /        \
+                                       each bit represents a word -> /...bits...\
+                                          in the dynamic heap
+
+## Marking with absolute addresses
+
+When marking, we pass an absolute pointer (i.e. address space relative), for speed.
+Internally the bitmap is kept in a (non-moving) blob behind the dynamic heap (DH).
+To efficiently mark the right bit in the bitmap, we maintain a pointer that points
+_before the start of the bitmap_ such that using the `/%8`-operation on the DH
+absolute word number will address the right bit:
+
+
+       +---- BITMAP_FORBIDDEN_PTR         +---- BITMAP_PTR
+       v                                  v
+       ,          (forbidden)             ,...................... bitmap ..........~~....,
+       |   heap_prefix_words / 8 bytes    |               heap_size / 32 bytes           |
+       |   get_bitmap_forbidden_size()    |                   BITMAP_SIZE                |
+       ^                                  ^                                              ^
+      /           ^^^^^^^^^^^              \                                             |
+     /            corresponds               \                 ^^^^^^^^^^^               /
+    /                                        \                corresponds              /
+   /                                          \                                       /
+  +---- Rts stack ----+---- Motoko statics ----+--------- Dynamic heap ----------~~--+
+                                               !
+                                               ! 32-byte aligned
+
+
+Debug assertions guard the forbidden bytes from access, as this area potentially
+physically overlaps with the Motoko dynamic heap.
+
+## The alignment caveat
+
+For this scheme to work, it is essential that the start of the DH is an address that
+is divisible by 32 (implying that `heap_prefix_words % 8 == 0`). Otherwise the
+`/%8`-operation on the DH's starting address will not yield the least significant bit
+in the BM, and thus the sweep operation will be off.
```

Comment:
```suggestion
is divisible by 32 (`heap_prefix_words % 8 == 0`). Otherwise the `address / 8`
and `address % 8` operations on the DH's starting address will not yield the
least significant bit in the BM.
```

Changes:

- "implying that" part gone, to keep it short
- Removed unfamiliar `%/` syntax
- "sweep operation will be off" we don't do sweep. This is important as sweep thouches unreachable objects, but we do compaction and touch reachable objects.
- Formatted with vim's `gq`